### PR TITLE
Run one worker on main thread

### PR DESCRIPTION
**Pull request by sberkun**
Followup to #344

On embedded platforms, we have a very limited number of threads (i.e. just two on the rp2040). Therefore, we would like the main thread to run a worker, rather than just sitting idle while other threads run workers. This PR proposes the simplest way of doing so.

One alternate approach is to make a list of threads, and a function that runs them all at once (essentially separating out the fork/join logic). I think it would look cleaner, but would involve a significant amount of extra code to create a "thread list" type. Since this "thread list" type would only be used once, I'm not sure if its worth the tradeoff.

Open to any other approaches as well.

### CODE REVIEW

File: core/threaded/reactor_threaded.c [hunk 1]
```diff
@@ -1161,23 +1150,48 @@ int lf_reactor_c_main(int argc, const char* argv[]) {
...
+            if (i == 0 && j == 0) {
+                // main thread worker
+                worker_thread_exit_status = main_thread_exit_status;
+            } else {
+                int failure = lf_thread_join(env->thread_ids[j], &worker_thread_exit_status);
+                if (failure) {
+                    lf_print_error("Failed to join thread listening for incoming messages: %s", strerror(failure));
+                }
+            }
         	if (worker_thread_exit_status != NULL) {
```
Comment (edwardalee) [inline core/threaded/reactor_threaded.c:1193]:
```suggestion
            if (worker_thread_exit_status != NULL) {
```

File: core/threaded/reactor_threaded.c [hunk 2]
```diff
@@ -1161,23 +1150,48 @@ int lf_reactor_c_main(int argc, const char* argv[]) {
...
-                lf_print_error("---- Worker %d reports error code %p", i, worker_thread_exit_status);
+                lf_print_error("---- Worker %d reports error code %p", j, worker_thread_exit_status);
                 ret = 1;
         	}
```
Comment (edwardalee) [inline core/threaded/reactor_threaded.c:1196]:
```suggestion
            }
```

**Review by edwardalee — APPROVED** [review]
This looks good to me.  If all tests pass and conflicts are resolved, I don't see any reason to not merge this.

**Comment by lhstrh** [comment]
@sberkun, it looks like this PR has stalled and is at risk of falling behind (esp. after merging #384). Shall we try to get this merged soon? There are some unresolved issues left to address...

**Comment by sberkun** [comment]
I was going to wait for #344 to get merged first, but since these PRs are relatively independent I'll just rebase these changes.

**Comment by sberkun** [comment]
Replaced by #437

**Pull request closed without merging**